Neste documento vamos analisar os dados da eleição municipal de são paulo, primeiro turno.
Os dados utilizado foram retirados do site https://www.tse.jus.br/hotsites/pesquisas-eleitorais/resultados_anos/2020.html

In [1]:
import pandas as pd

In [9]:
#esse arquivo é bem grande, meu computador leu com dificuldades. Não consegui abrir o arquivo pelo bloco de notas
data = pd.read_csv('C:/Users/juacysantos/Downloads/bweb_1t_SP_181120201549/bweb_1t_SP_181120201549.csv',
                   error_bad_lines=False,
                  encoding='latin1',
                  sep=';')

In [10]:
data

DT_GERACAO HH_GERACAO  ANO_ELEICAO  CD_TIPO_ELEICAO  \
0        18/11/2020   15:49:53         2020                0   
1        18/11/2020   15:49:53         2020                0   
2        18/11/2020   15:49:53         2020                0   
3        18/11/2020   15:49:53         2020                0   
4        18/11/2020   15:49:53         2020                0   
...             ...        ...          ...              ...   
9433995  18/11/2020   15:49:53         2020                0   
9433996  18/11/2020   15:49:53         2020                0   
9433997  18/11/2020   15:49:53         2020                0   
9433998  18/11/2020   15:49:53         2020                0   
9433999  18/11/2020   15:49:53         2020                0   

           NM_TIPO_ELEICAO  CD_PLEITO   DT_PLEITO  NR_TURNO  CD_ELEICAO  \
0        Eleição Ordinária        304  15/11/2020         1         426   
1        Eleição Ordinária        304  15/11/2020         1         426   
2        Eleição Ordinária        304  15/11/2020         1         426   
3        Eleição Ordinária        304  15/11/2020         1         426   
4        Eleição Ordinária        304  15/11/2020         1         426   
...                    ...        ...         ...       ...         ...   
9433995  Eleição Ordinária        304  15/11/2020         1         426   
9433996  Eleição Ordinária        304  15/11/2020         1         426   
9433997  Eleição Ordinária        304  15/11/2020         1         426   
9433998  Eleição Ordinária        304  15/11/2020         1         426   
9433999  Eleição Ordinária        304  15/11/2020         1         426   

                       DS_ELEICAO  ... CD_FLASHCARD_URNA_EFETIVADA  \
0        Eleições Municipais 2020  ...                    E0BA7264   
1        Eleições Municipais 2020  ...                    E0BA7264   
2        Eleições Municipais 2020  ...                    E0BA7264   
3        Eleições Municipais 2020  ...                    E0BA7264   
4        Eleições Municipais 2020  ...                    E0BA7264   
...                           ...  ...                         ...   
9433995  Eleições Municipais 2020  ...                    36C93684   
9433996  Eleições Municipais 2020  ...                    36C93684   
9433997  Eleições Municipais 2020  ...                    36C93684   
9433998  Eleições Municipais 2020  ...                    36C93684   
9433999  Eleições Municipais 2020  ...                    36C93684   

         DT_CARGA_URNA_EFETIVADA DS_CARGO_PERGUNTA_SECAO  DS_AGREGADAS  \
0            03/11/2020 11:19:00                  11 - 1        #NULO#   
1            03/11/2020 11:19:00                  11 - 1        #NULO#   
2            03/11/2020 11:19:00                  11 - 1        #NULO#   
3            03/11/2020 11:19:00                  11 - 1        #NULO#   
4            03/11/2020 11:19:00                  11 - 1        #NULO#   
...                          ...                     ...           ...   
9433995      08/11/2020 12:08:00                 13 - 73        #NULO#   
9433996      08/11/2020 12:08:00                 13 - 73        #NULO#   
9433997      08/11/2020 12:08:00                 13 - 73        #NULO#   
9433998      08/11/2020 12:08:00                 13 - 73        #NULO#   
9433999      08/11/2020 12:08:00                 13 - 73        #NULO#   

                 DT_ABERTURA      DT_ENCERRAMENTO  QT_ELEITORES_BIOMETRIA_NH  \
0        15/11/2020 07:00:00  15/11/2020 17:02:46                          0   
1        15/11/2020 07:00:00  15/11/2020 17:02:46                          0   
2        15/11/2020 07:00:00  15/11/2020 17:02:46                          0   
3        15/11/2020 07:00:00  15/11/2020 17:02:46                          0   
4        15/11/2020 07:00:00  15/11/2020 17:02:46                          0   
...                      ...                  ...                        ...   
9433995  15/11/2020 07:00:00  15/11/2020 17:03:43

In [28]:
#verificando a urna na secção em que fui mesário durante essas eleições
pd.set_option('display.max_columns', None)
data[(data['NR_ZONA']==404) & (data['NR_SECAO']==190)]

DT_GERACAO HH_GERACAO  ANO_ELEICAO  CD_TIPO_ELEICAO  \
8601208  18/11/2020   15:49:53         2020                0   
8601209  18/11/2020   15:49:53         2020                0   
8601210  18/11/2020   15:49:53         2020                0   
8601211  18/11/2020   15:49:53         2020                0   
8601212  18/11/2020   15:49:53         2020                0   
...             ...        ...          ...              ...   
8601319  18/11/2020   15:49:53         2020                0   
8601320  18/11/2020   15:49:53         2020                0   
8601321  18/11/2020   15:49:53         2020                0   
8601322  18/11/2020   15:49:53         2020                0   
8601323  18/11/2020   15:49:53         2020                0   

           NM_TIPO_ELEICAO  CD_PLEITO   DT_PLEITO  NR_TURNO  CD_ELEICAO  \
8601208  Eleição Ordinária        304  15/11/2020         1         426   
8601209  Eleição Ordinária        304  15/11/2020         1         426   
8601210  Eleição Ordinária        304  15/11/2020         1         426   
8601211  Eleição Ordinária        304  15/11/2020         1         426   
8601212  Eleição Ordinária        304  15/11/2020         1         426   
...                    ...        ...         ...       ...         ...   
8601319  Eleição Ordinária        304  15/11/2020         1         426   
8601320  Eleição Ordinária        304  15/11/2020         1         426   
8601321  Eleição Ordinária        304  15/11/2020         1         426   
8601322  Eleição Ordinária        304  15/11/2020         1         426   
8601323  Eleição Ordinária        304  15/11/2020         1         426   

                       DS_ELEICAO SG_UF  CD_MUNICIPIO NM_MUNICIPIO  NR_ZONA  \
8601208  Eleições Municipais 2020    SP         71072    SÃO PAULO      404   
8601209  Eleições Municipais 2020    SP         71072    SÃO PAULO      404   
8601210  Eleições Municipais 2020    SP         71072    SÃO PAULO      404   
8601211  Eleições Municipais 2020    SP         71072    SÃO PAULO      404   
8601212  Eleições Municipais 2020    SP         71072    SÃO PAULO      404   
...                           ...   ...           ...          ...      ...   
8601319  Eleições Municipais 2020    SP         71072    SÃO PAULO      404   
8601320  Eleições Municipais 2020    SP         71072    SÃO PAULO      404   
8601321  Eleições Municipais 2020    SP         71072    SÃO PAULO      404   
8601322  Eleições Municipais 2020    SP         71072    SÃO PAULO      404   
8601323  Eleições Municipais 2020    SP         71072    SÃO PAULO      404   

         NR_SECAO  NR_LOCAL_VOTACAO  CD_CARGO_PERGUNTA DS_CARGO_PERGUNTA  \
8601208       190              1228                 11          Prefeito   
8601209       190              1228                 11          Prefeito   
8601210       190              1228                 11          Prefeito   
8601211       190              1228                 11          Prefeito   
8601212       190              1228                 11          Prefeito   
...           ...               ...                ...               ...   
8601319       190              1228                 13          Vereador   
8601320       190              1228                 13          Vereador   
8601321       190              1228                 13          Vereador   
8601322       190              1228                 13          Vereador   
8601323       190              1228                 13          Vereador   

         NR_PARTIDO SG_PARTIDO                                NM_PARTIDO  \
8601208          45       PSDB   Partido da Social Democracia Brasileira   
8601209          -1     #NULO#                                    #NULO#   
8601210          40        PSB             Partido Socialista Brasileiro   
8601211          28       PRTB  Partido Renovador Trabalhista Brasileiro   
8601212          13         PT                 Partido dos Trabalhadores   
...             ...  

In [31]:
#filtrando as linhas apenas do municipio de são paulo e relacionado ao cargo de prefeito
data = data[(data['NM_MUNICIPIO']=='SÃO PAULO') & (data['DS_CARGO_PERGUNTA']=='Prefeito')]

In [41]:
data

DT_GERACAO HH_GERACAO  ANO_ELEICAO  CD_TIPO_ELEICAO  \
0        18/11/2020   15:49:53         2020                0   
1        18/11/2020   15:49:53         2020                0   
2        18/11/2020   15:49:53         2020                0   
3        18/11/2020   15:49:53         2020                0   
4        18/11/2020   15:49:53         2020                0   
...             ...        ...          ...              ...   
9355098  18/11/2020   15:49:53         2020                0   
9355099  18/11/2020   15:49:53         2020                0   
9355100  18/11/2020   15:49:53         2020                0   
9355101  18/11/2020   15:49:53         2020                0   
9355102  18/11/2020   15:49:53         2020                0   

           NM_TIPO_ELEICAO  CD_PLEITO   DT_PLEITO  NR_TURNO  CD_ELEICAO  \
0        Eleição Ordinária        304  15/11/2020         1         426   
1        Eleição Ordinária        304  15/11/2020         1         426   
2        Eleição Ordinária        304  15/11/2020         1         426   
3        Eleição Ordinária        304  15/11/2020         1         426   
4        Eleição Ordinária        304  15/11/2020         1         426   
...                    ...        ...         ...       ...         ...   
9355098  Eleição Ordinária        304  15/11/2020         1         426   
9355099  Eleição Ordinária        304  15/11/2020         1         426   
9355100  Eleição Ordinária        304  15/11/2020         1         426   
9355101  Eleição Ordinária        304  15/11/2020         1         426   
9355102  Eleição Ordinária        304  15/11/2020         1         426   

                       DS_ELEICAO SG_UF  CD_MUNICIPIO NM_MUNICIPIO  NR_ZONA  \
0        Eleições Municipais 2020    SP         71072    SÃO PAULO        1   
1        Eleições Municipais 2020    SP         71072    SÃO PAULO        1   
2        Eleições Municipais 2020    SP         71072    SÃO PAULO        1   
3        Eleições Municipais 2020    SP         71072    SÃO PAULO        1   
4        Eleições Municipais 2020    SP         71072    SÃO PAULO        1   
...                           ...   ...           ...          ...      ...   
9355098  Eleições Municipais 2020    SP         71072    SÃO PAULO      422   
9355099  Eleições Municipais 2020    SP         71072    SÃO PAULO      422   
9355100  Eleições Municipais 2020    SP         71072    SÃO PAULO      422   
9355101  Eleições Municipais 2020    SP         71072    SÃO PAULO      422   
9355102  Eleições Municipais 2020    SP         71072    SÃO PAULO      422   

         NR_SECAO  NR_LOCAL_VOTACAO  CD_CARGO_PERGUNTA DS_CARGO_PERGUNTA  \
0               1              1015                 11          Prefeito   
1               1              1015                 11          Prefeito   
2               1              1015                 11          Prefeito   
3               1              1015                 11          Prefeito   
4               1              1015                 11          Prefeito   
...           ...               ...                ...               ...   
9355098       402              1392                 11          Prefeito   
9355099       402              1392                 11          Prefeito   
9355100       402              1392                 11          Prefeito   
9355101       402              1392                 11          Prefeito   
9355102       402              1392                 11          Prefeito   

         NR_PARTIDO    SG_PARTIDO                               NM_PARTIDO  \
0                -1        #NULO#                                   #NULO#   
1                13            PT                Partido dos Trabalhadores   
2                18          REDE                    Rede Sustentabilidade   
3                40           PSB            Partido Socialista Brasileiro   
4                45          PSDB  Partido da Social Democracia Brasileira   
...      

Pretendo fazer uma analise por zona/bairro para facilitar a compreensão irei utilizar os dados disponível em https://pt.wikipedia.org/wiki/Zonas_eleitorais_da_cidade_de_S%C3%A3o_Paulo
Para relacionar cada sessão com seu bairro

In [34]:
zonas = pd.read_csv('C:/Users/juacysantos/Documents/GitHub/training-data-exploration/zonas_eleitorais_sp.csv')

In [35]:
zonas

zona                   bairro
0      1               Bela Vista
1      2                 Perdizes
2      3           Santa Ifigênia
3      4                    Mooca
4      5          Jardim Paulista
5      6             Vila Mariana
6     20               Valo Velho
7    246              Santo Amaro
8    247      São Miguel Paulista
9    248                 Itaquera
10   249                  Santana
11   250                     Lapa
12   251                Pinheiros
13   252          Penha de França
14   253                  Tatuapé
15   254               Vila Maria
16   255               Casa Verde
17   256                 Tucuruvi
18   257            Vila Prudente
19   258             Indianópolis
20   259                    Saúde
21   260                 Ipiranga
22   280        Capela do Socorro
23   320                Jabaquara
24   325                 Pirituba
25   326       Ermelino Matarazzo
26   327       Nossa Senhora do Ó
27   328              Campo Limpo
28   346                  Butantã
29   347             Vila Matilde
30   348             Vila Formosa
31   349                   Jaçanã
32   350                Sapopemba
33   351            Cidade Ademar
34   352           Itaim Paulista
35   353               Guaianases
36   371                   Grajaú
37   372              Piraporinha
38   373            Capão Redondo
39   374              Rio Pequeno
40   375               São Mateus
41   376              Brasilândia
42   381              Parelheiros
43   389                    Perus
44   390                 Cangaíba
45   392               Ponte Rasa
46   397            Jardim Helena
47   398               Vila Jacuí
48   403                  Jaraguá
49   404        Cidade Tiradentes
50   405  Conjunto José Bonifácio
51   408          Jardim São Luís
52   413                  Cursino
53   417          Parque do Carmo
54   418                 Pedreira
55   420             Vila Sabrina
56   421          Teotônio Vilela
57   422         Lauzane Paulista

In [39]:
#quantas zonas eleitorais temos?
len(pd.unique(data['NR_ZONA']))

58

In [48]:
#quantas sessões temos por zona eleitoral?
data_agr = data.groupby(data['NR_ZONA'])
data_agr = data_agr.agg({'NR_SECAO':'nunique'})
data_agr

NR_SECAO
NR_ZONA          
1             373
2             437
3             189
4             392
5             305
6             387
20            391
246           287
247           278
248           382
249           284
250           418
251           296
252           287
253           545
254           385
255           453
256           435
257           404
258           447
259           371
260           433
280           405
320           437
325           245
326           276
327           296
328           427
346           342
347           422
348           258
349           323
350           321
351           325
352           503
353           457
371           478
372           665
373           326
374           629
375           610
376           603
381           567
389           485
390           303
392           328
397           384
398           263
403           271
404           459
405           268
408           393
413           360
417           300
418           464
420           308
421           353
422           366

In [46]:
#confirmando que a sessão 404 possui 459 sessões
len(pd.unique(data[data['NR_ZONA']==404]['NR_SECAO']))

459

Com esses dados das sessões por zona, conseguimos estimar quantos mesários participaram das eleições no municipio de são paulo. Como temos 4 mesários por sessão (Presidente, 1° e 2° mesários e secretário).

In [54]:
sum(data_agr['NR_SECAO'])*4

89596

Levando isso em conta, tivemos então 89596 mesários atuando no municipio de São Paulo. Aproximadamente 90 mil pessoas.

Contaremos agora a quantidade de votos por zona dos candidatos a prefeito e também a quantidade de votos brancos, nulos e abstenções. Assim conseguiremos analisar cada zona eleitoral.

In [66]:
pd.set_option('display.max_rows', None)
data_agr = data.groupby([data['NR_ZONA'], data['NM_VOTAVEL']])
data_agr = data_agr.agg({'QT_VOTOS':'sum'})
data_agr.sort_values(['NR_ZONA','QT_VOTOS'], ascending=False)



QT_VOTOS
NR_ZONA NM_VOTAVEL                         
422     BRUNO COVAS                   28271
        GUILHERME BOULOS              16578
        MÁRCIO FRANÇA                 13103
        Nulo                          11075
        ARTHUR DO VAL MAMÃE FALEI      9340
        CELSO RUSSOMANNO               9238
        Branco                         6456
        JILMAR TATTO                   5032
        JOICE HASSELMANN               2372
        ANDREA MATARAZZO               1536
        MARINA HELOU                    332
        LEVY FIDELIX                    215
        ORLANDO SILVA                   117
        VERA                             48
        ANTÔNIO CARLOS                   10
421     BRUNO COVAS                   24312
        GUILHERME BOULOS              15913
        Nulo                          12466
        JILMAR TATTO                  11541
        MÁRCIO FRANÇA                 11356
        CELSO RUSSOMANNO              10617
        Branco                         7526
        ARTHUR DO VAL MAMÃE FALEI      7183
        JOICE HASSELMANN               1637
        ANDREA MATARAZZO                948
        ORLANDO SILVA                   261
        LEVY FIDELIX                    224
        MARINA HELOU                    108
        VERA                             34
        ANTÔNIO CARLOS                   15
420     BRUNO COVAS                   24576
        GUILHERME BOULOS              11777
        MÁRCIO FRANÇA                 10738
        Nulo                           9674
        CELSO RUSSOMANNO               9114
        ARTHUR DO VAL MAMÃE FALEI      7103
        Branco                         5591
        JILMAR TATTO                   3130
        JOICE HASSELMANN               1418
        ANDREA MATARAZZO                920
        LEVY FIDELIX                    200
        MARINA HELOU                    180
        ORLANDO SILVA                   174
        VERA                             40
        ANTÔNIO CARLOS                   11
418     BRUNO COVAS                   35840
        GUILHERME BOULOS              19113
        Nulo                          15306
        JILMAR TATTO                  14853
        MÁRCIO FRANÇA                 13703
        CELSO RUSSOMANNO              12919
        ARTHUR DO VAL MAMÃE FALEI      9929
        Branco                         9367
        JOICE HASSELMANN               2007
        ANDREA MATARAZZO               1381
        MARINA HELOU                    249
        LEVY FIDELIX                    238
        ORLANDO SILVA                   213
        VERA                             69
        ANTÔNIO CARLOS                    9
417     BRUNO COVAS                   20512
        GUILHERME BOULOS              15446
        MÁRCIO FRANÇA                 11532
        Nulo                          10875
        CELSO RUSSOMANNO               8432
        ARTHUR DO VAL MAMÃE FALEI      7392
        JILMAR TATTO                   6075
        Branco                         5899
        JOICE HASSELMANN               1448
        ANDREA MATARAZZO                904
        MARINA HELOU                    169
        LEVY FIDELIX                    167
        ORLANDO SILVA                   148
        VERA                             64
        ANTÔNIO CARLOS                    6
413     BRUNO COVAS                   30731
        GUILHERME BOULOS              17210
        MÁRCIO FRANÇA                 13698
        Nulo                          10838
        ARTHUR DO VAL MAMÃE FALEI      9275
        CELSO RUSSOMANNO               8665
        JILMAR TATTO                   6395
        Branco                         5679
        JOICE HASSELMANN               1604
        ANDREA MATARAZZO               1138
        MARINA HELOU                    310
        LEVY FIDELIX                    211
        ORLANDO SILVA                   182
        VERA                             50
        ANTÔNIO CARLOS         

In [69]:
pd.set_option('display.max_rows', 10)
abstencoes = data[['NR_ZONA', 'NR_SECAO', 'QT_APTOS', 'QT_COMPARECIMENTO', 'QT_ABSTENCOES']]
abstencoes = abstencoes.drop_duplicates()
abstencoes

NR_ZONA  NR_SECAO  QT_APTOS  QT_COMPARECIMENTO  QT_ABSTENCOES
0              1         1       386                241            145
135            1         2       450                288            162
270            1         3       452                281            171
419            1         4       449                269            180
560            1         5       450                293            157
...          ...       ...       ...                ...            ...
9354719      422       396       400                302             98
9354839      422       397       378                253            125
9354951      422       400       228                184             44
9355029      422       401       133                111             22
9355094      422       402       100                 82             18

[22399 rows x 5 columns]

In [72]:
abstencoes = abstencoes.groupby('NR_ZONA').agg({'QT_ABSTENCOES':'sum', 'QT_APTOS':'sum'})
abstencoes

QT_ABSTENCOES  QT_APTOS
NR_ZONA                         
1                59518    153402
2                61557    180555
3                29158     76816
4                53079    160222
5                46976    122253
...                ...       ...
417              33627    122696
418              53374    188570
420              35318    119964
421              37553    141694
422              40795    144518

[58 rows x 2 columns]

In [77]:
#calculando o percentual de abstenções por sessão eleitoral
abstencoes['percentual'] = abstencoes.apply(lambda x:round((x['QT_ABSTENCOES']/x['QT_APTOS'])*100,2), axis=1)
abstencoes

QT_ABSTENCOES  QT_APTOS  percentual
NR_ZONA                                     
1                59518    153402       38.80
2                61557    180555       34.09
3                29158     76816       37.96
4                53079    160222       33.13
5                46976    122253       38.43
...                ...       ...         ...
417              33627    122696       27.41
418              53374    188570       28.30
420              35318    119964       29.44
421              37553    141694       26.50
422              40795    144518       28.23

[58 rows x 3 columns]

In [ ]:
#votos nulos e brancos por zona
#fazer um grafico mostrando a posição do candidato por zona eleitoral
#contar o numero de abstenções por zona eleitoral